cmd/tempo/app/modules.go

```diff
@@ -158,8 +158,11 @@ func (t *App) initQuerier() (services.Service, error) {
 }
 
 func (t *App) initQueryFrontend() (services.Service, error) {
-	var err error
+	if t.cfg.Frontend.QueryShards < frontend.MinQueryShards || t.cfg.Frontend.QueryShards > frontend.MaxQueryShards {
```

Comment:
any particular reason we're capping at 256?

Reply:
As of now we're using only the first 8 bits of the blockID for sharding. We could extend that but it seems like a reasonable limit for number of shards